src/chkstat.cpp

```diff
@@ -21,1365 +21,1246 @@
+static inline std::string& stripQuotes(std::string &s)
 {
-    return (c == '"' || c == '\'');
+    return strip(s, [](char c) { return c == '"' || c == '\''; });
 }
 
-int
-parse_sysconf(const char* file)
+bool Chkstat::parseSysconfig()
 {
-    FILE* fp;
-    char line[PATH_MAX];
-    char* p;
+    const auto file = m_config_root_path.getValue() + "/etc/sysconfig/security";
+    std::ifstream fs(file);
 
-    if ((fp = fopen(file, "r")) == 0)
+    if (!fs)
     {
-        fprintf(stderr, "error opening: %s: %s\n", file, strerror(errno));
-        return 0;
+        // NOTE: the original code tolerates this situation and continues
+        std::cerr << "error opening " << file << ": " << std::strerror(errno) << std::endl;
+        return true;
     }
 
-    while (readline(fp, line, sizeof(line)))
+    std::string line;
+    std::string key;
+    std::string value;
+    size_t linenr = 0;
+    bool check_permissions = true;
+
+    while (std::getline(fs, line))
     {
-        if (!*line)
-            continue;
+        linenr++;
+        strip(line);
 
-        for (p = line; *p == ' '; ++p);
+        if (line.empty() || line[0] == '#')
+            continue;
 
-        if (!*p || *p == '#')
+        const auto sep = line.find_first_of('=');
+        if (sep == line.npos)
+        {
+            // syntax error?
+            std::cerr << file << ":" << linenr << ": parse error in '" << line << "'" << std::endl;
             continue;
+        }
 
-        if (!strncmp(p, "PERMISSION_SECURITY=", 20))
+        key = line.substr(0, sep);
+        value = line.substr(sep + 1);
+        value = stripQuotes(value);
+
+        if (key == "PERMISSION_SECURITY")
         {
-            if (force_level)
+            if (m_force_level_list.isSet())
+                // explicit levels are specified on the command line
                 continue;
 
-            p+=20;
-            if (isquote(*p))
-                ++p;
-            p = strtok(p, " ");
-            if (p && !isquote(*p))
+            std::vector<std::string> profiles;
+            splitWords(value, profiles);
+
+            for (const auto &profile: profiles)
             {
-                do
+                if (profile != "local" && !profile.empty())
                 {
-                    if (isquote(p[strlen(p)-1]))
-                    {
-                        p[strlen(p)-1] = '\0';
-                    }
-                    if (*p && strcmp(p, "local"))
-                        add_level(p);
+                    addProfile(profile);
                 }
-                while ((p = strtok(NULL, " ")));
             }
         }
-        else if (!strncmp(p, "CHECK_PERMISSIONS=", 18))
+        // this setting was last seen in the config file template in SLE-11
+        // but we still support it in the code. The logic behind it is quite
+        // complex since it is crossed with the command line settings (also
+        // see logic after the while loop).
```

Comment:
Something like `interlocked`. Meaning the application of this setting depends on the values of command line switches as can be seen a few lines below.

Actually I don't think parsing this item still makes sense but for the initial refactoring I'd like to change the behaviour and feature set very little. So that we have a defined state of the code we're it's become more maintainable. And from there we can start throwing out old stuff or implement new stuff.

Reply:
Can you reword the comment then?

Agreed about removing this at some point. Should we keep track of this somewhere? An issue for every one of these little things seems excessive, but maybe with "REMOVEME" comments, or similar?

Reply:
If I'm not completely mistaken then `crossed` is also valid here but if it is hard to grasp then I'll change it to ´interlocked`. I'll add a REMOVEME, too.